store/cachekv/btree.go

```diff
@@ -0,0 +1,72 @@
+package cachekv
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/tidwall/btree"
+)
+
+var errKeyEmpty = errors.New("key cannot be empty")
+
+type BTree struct {
```

Comment:
could we make this implementation internal to cachekv by making it private or putting it internal? don't want to add things to the public api that aren't not needed. 

Also can we put a go doc on why this lives here instead of memdb?

Reply:
> putting it internal

how to put it internal? make a `internal` directory?

Reply:
yea, sorry internal directory

Reply:
done

Reply:
> Also can we put a go doc on why this lives here instead of memdb?

done